cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -37,37 +42,56 @@ const COMPLIANCE_STATUS_MAP = new Map([
 
 /**
  * Inserts all rows into the violations table.
+ * @param db: Database to use
+ * @param rows: Rows to write into the db
  */
-async function insertRows(db: ConnectionPool, rows: ViolationsTableRow[]): Promise<any[]> {
-  return db.query(sql`INSERT INTO epa_violations (violation_id,
-                                                  pws_id,
-                                                  violation_code,
-                                                  compliance_status,
-                                                  start_date)
-                      VALUES ${sql.join(
-                        rows.map((row: ViolationsTableRow) => {
-                          return sql`(${row.violation_id}, ${row.pws_id},
-                                                 ${row.violation_code},
-                                                 ${row.compliance_status},
-                                                 ${row.start_date})`;
-                        }),
-                        ',',
-                      )} ON CONFLICT (violation_id) DO UPDATE;`);
+async function insertRows(db: PoolClient, rows: ViolationsTableRow[]): Promise<QueryArrayResult> {
+  const valuesToInsert: any[][] = [];
+
+  for (let row of rows) {
+    valuesToInsert.push([
+      row.violation_id,
+      row.pws_id,
+      row.violation_code,
+      row.compliance_status,
+      row.start_date,
+    ]);
+  }
+
+  // Format function below needs these to be %s (string literals) or else
+  // it produces invalid geometries.
+  const insertIntoStatement =
+    'INSERT INTO epa_violations (violation_id, pws_id,violation_code, compliance_status, start_date) ' +
+    'VALUES %L ON CONFLICT (violation_id) DO NOTHING';
+
+  try {
+    await db.query('BEGIN');
+    const queryResult = await db.query(format(insertIntoStatement, valuesToInsert), []);
+    await db.query('COMMIT');
+    return queryResult;
+  } catch (error) {
+    await db.query('ROLLBACK');
+    throw error;
+  }
 }
 
 /**
- * Reads the S3 CSV file and the number of rows successfully written.
+ * Reads the S3 file and the number of rows successfully written.
+ * @param s3Params: Params that identity the s3 bucket
+ * @param db: Database to write to.
+ * @param startIndex: The row to begin writes with
+ * @param numberOfRowsToWrite: The number of entries to write to the db
  */
 function parseS3IntoViolationsTableRow(
   s3Params: AWS.S3.GetObjectRequest,
-  db: ConnectionPool,
+  db: PoolClient,
   startIndex: number,
```

Comment:
How do you know what `startIndex` to use?

Reply:
Based on rows in your db

Reply:
[link to my other comment](https://github.com/BlueConduit/open-data-platform/pull/43#pullrequestreview-1011168858) about how rows don't insert in order, so you can't guarantee that N rows in the DB means the first N rows from the file.

I don't think `startIndex` is needed once we split the file anyway. But we can cut that out in a future PR, since I don't think it's really used yet.